### DISCUSSION TITLE
Make Redirects available in Static HTML Export

### DISCUSSION THREAD

**Discussion by tkw1536** [vercel/next.js#17322] [Ideas]
# Feature request

Recently [Redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) have been added to NextJS. However these are only available on NodeJS environments, making them less useful in cases where the resulting application is served via a static webserver. 

I would like to request support for redirects in the Static HTML Export. 

## Describe alternatives you've considered

There are ways of achieving a similar effect to server-side redirects. In particular, I have considered two alternatives. 

Suppose I want to redirect `/source/` to `/target/`, then both of these require me to create a dummy page `pages/source.jsx`.
In my view, this defeats the purpose of having redirects in `next.config.js` in the first place. 

__Version 1:__ 

Add a `componentDidMount` hook that uses the Router to immediately replace the URL. 

```js
componentDidMount() {
    Router.replace(`/target/`);
}
```
__Version 2:__ 

Inject a `meta http-equiv="refresh"` tag into the page. Concretely, inside the `render` method add:

```js
<Head>
    <meta http-equiv="refresh" content="0; url=/target/" />
</Head>
```

## Describe the solution you'd like

I would like a solution where for every redirect page a static html file with a tag similar to version 2 above could be generated. 
This would have the advantage that no compiled JavaScript Code would be required on those pages, making them very simple to generate.

**Comment by rasovica** [comment]
I would love this too!

**Comment by varanauskas** [comment]
I've came up with a different possible solution for this:
https://github.com/varanauskas/nexered

It works if there is support for HTTP level redirects where you host the page.
For example it is currently implemented for Cloudflare Pages (that supports redirects defined in _redirects file, similar to Netlify)
If we can parse the existing redirects in `next.config.js` then we can generate this file for third party hosters
(possibly also Apache htaccess files and Nginx config files)

This implementation is hosting provider dependent, however, it works on the HTTP level (good for SEO and speed)

if there is interest I am thinking about adding other provider support for it (Netlify, Apache .htaccess, Nginx...)

**Reply by tkw1536** [reply]
That looks cool.
Although I am using GitHub pages and they do not support any kind of redirects.

**Comment by alexander-williamson** [comment]
👍 Upvote from me on this feature please! I use SSG all the time, having ported everything over from Gatsby to Next.js
